### PR TITLE
Handling timeout exception for monitor UNKNOWN error

### LINKED ISSUE #8762: Monitor cluster health indicator reports UNKNOWN when timed out
### Description

Monitor cluster health indicator uses rest api network stake endpoint to determine if there's db issue and should report `DOWN` when it happens.

However, it seems like when the rest service is not available (no pods ready), sporadically, it times out and report `UNKNOWN` instead.

```
2024-07-12T21:19:16.660Z ERROR parallel-35 c.h.m.m.h.ClusterHealthIndicator Network stake status is UNKNOWN with error: Did not observe any item or terminal signal within 5000ms in 'flatMap' (and no fallback has been configured) 
```

### Steps to reproduce

Check monitor pod logs

### Additional context

_No response_

### Hedera network

other

### Version

v0.108.0

### Operating system

None

### PR DESCRIPTION
**Pull request by mgoelswirlds**
**Description**:

Handling timeout exception for monitor UNKNOWN error

This PR modifies 
* Looks for timeout exception in ClusterHealthIndicator.

**Related issue(s)**:

Fixes #8762

**Notes for reviewer**:
This has not been reproduced. Would need help reproducing it.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by jascks — DISMISSED** [review]
LGTM

**Review by jnels124 — APPROVED** [review]
LGTM

File: hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/health/ClusterHealthIndicator.java [hunk 1]
```diff
@@ -100,7 +101,8 @@ private Mono<Health> restNetworkStakeHealth() {
                     var status = Status.UNKNOWN;
                     // Connection issue can be caused by database being down, since the rest API service will become
                     // unavailable eventually
-                    if (ExceptionUtils.getRootCause(e) instanceof ConnectException) {
+                    if (ExceptionUtils.getRootCause(e) instanceof ConnectException
+                            || ExceptionUtils.getRootCause(e) instanceof TimeoutException) {
```
Comment (steven-sheehy) [inline hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/health/ClusterHealthIndicator.java:105]:
```suggestion
                    var rootCause = ExceptionUtils.getRootCause(e);
                    if (rootCause instanceof ConnectException || rootCause instanceof TimeoutException) {
```

File: hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java [hunk 2]
```diff
@@ -100,6 +101,20 @@ void restNetworkStakeConnectException() {
...
 
+    @SneakyThrows
+    @Test
+    void restNetworkStakeTimeoutException() {
+        var exception = new WebClientRequestException(
+                new TimeoutException("Timeout occured"),
+                HttpMethod.GET,
+                new URI("http://localhost/api/v1/network/stake"),
+                HttpHeaders.EMPTY);
+        when(restApiClient.getNetworkStakeStatusCode()).thenReturn(Mono.error(exception));
```
Comment (steven-sheehy) [inline hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java:112]: This doesn't actually test the flux `timeout(5)` is hit. Instead it just fakes a different timeout exception. We should use virtual time and have restApiClient take longer than 5 virtual seconds.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by xin-hedera** [comment]
was able to reproduce it to it throws a `java.util.concurrent.TimeoutException` exception, though there are changes requested.

File: hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/health/ClusterHealthIndicator.java (hunk 1 above)
Comment (jascks) [reply hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/health/ClusterHealthIndicator.java]: Done

File: hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java (hunk 2 above)
Comment (jascks) [reply hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java]: Done

File: hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java [hunk 3]
```diff
@@ -47,11 +48,14 @@
 import org.springframework.web.reactive.function.client.WebClientRequestException;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
 
 @ExtendWith(MockitoExtension.class)
 @MockitoSettings(strictness = Strictness.LENIENT)
 class ClusterHealthIndicatorTest {
 
+    private static final Duration WAIT = Duration.ofSeconds(10L);
```
Comment (steven-sheehy) [inline hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java:57]: Unused

Comment (jascks) [reply hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/health/ClusterHealthIndicatorTest.java]: So close... Done!

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**